src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTargetChunks.java

```diff
@@ -271,7 +271,7 @@ private boolean isExecutableKnitrEngine(String engine)
    
    // runnable engines within the R Notebook mode
    private static final String RE_RUNNABLE_ENGINES =
-         "r|rscript|rcpp|python|ruby|perl|bash|sh|stan|sql|";
+         "r|rscript|rcpp|python|ruby|perl|bash|sh|stan|sql|d3|";
    
```

Comment:
I wonder what additional measures we should take to ensure the r2d3 package is available here. A few options:

1) Add r2d3 to the list of packages we automagically install alongside rmarkdown (we can do this using an embedded package until we have r2d3 on CRAN)

2) Prompt to install r2d3 whenever a user attempts to run a d3 chunk within a notebook

Or I guess we could leave things as-is and just have knit fail for the user if they don't have r2d3 installed (I'm presuming that in this case interactive execution from a notebook would still work).

Reply:
FWIW, we take approach (2) with `reticulate` -- any time you take some IDE action that requires `reticulate`, we prompt for its installation.

As an aside, we now (in theory) support _all_ engines registered to `knitr`, so we might consider synchronizing that list of engines between the server + client sides. This would allow users to define custom engines that automagically work in the IDE, beyond just `d3`.

Perhaps not for this PR but something worth thinking about?